# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webscheduling/WebTaskSchedulerWorker.h
# Commit: be6c223e63f0
# Full Hash: be6c223e63f0a0c8ef2428df42321d59e4c0f7c6
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-02-25 09:30:56
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Implement scheduler.yield r=jjaschke,dom-core,webidl,smaug
#   
#   Spec: https://wicg.github.io/scheduling-apis/#dom-scheduler-yield
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D234232
# ==============================================================================

diff -r 121bf804d2a6 -r be6c223e63f0 dom/webscheduling/WebTaskSchedulerWorker.h
--- a/dom/webscheduling/WebTaskSchedulerWorker.h	Mon Feb 24 19:31:26 2025 +0000
+++ b/dom/webscheduling/WebTaskSchedulerWorker.h	Mon Feb 24 19:31:27 2025 +0000
@@ -20,8 +20,7 @@
 
 class WebTaskWorkerRunnable final : public WorkerSameThreadRunnable {
  public:
-  WebTaskWorkerRunnable(WorkerPrivate* aWorkerPrivate,
-                        WebTaskSchedulerWorker* aSchedulerWorker);
+  explicit WebTaskWorkerRunnable(WebTaskSchedulerWorker* aSchedulerWorker);
 
   MOZ_CAN_RUN_SCRIPT_BOUNDARY
   bool WorkerRun(JSContext* aCx, WorkerPrivate* aWorkerPrivate) override;
@@ -42,8 +41,9 @@
  private:
   ~WebTaskSchedulerWorker() = default;
 
-  nsresult SetTimeoutForDelayedTask(WebTask* aTask, uint64_t aDelay) override;
-  bool DispatchEventLoopRunnable() override;
+  nsresult SetTimeoutForDelayedTask(WebTask* aTask, uint64_t aDelay,
+                                    EventQueuePriority aPriority) override;
+  bool DispatchEventLoopRunnable(EventQueuePriority aPriority) override;
 
   RefPtr<StrongWorkerRef> mWorkerRef;
   bool mWorkerIsShuttingDown{false};